### PR TITLE
Feature/cc 20 jwt and password encode

### PR DESCRIPTION
**Pull request by leonandhard**
add security filter and jwts

### CODE REVIEW

File: src/main/java/com/courtcanva/ccfranchise/jwts/JwtConfig.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpHeaders;
+
+import javax.crypto.SecretKey;
+
+
+@Configuration
+@Data
```
[the remark this answers is not in the document]
Comment (leonandhard) [reply src/main/java/com/courtcanva/ccfranchise/jwts/JwtConfig.java]: just need setter， thx

File: src/main/java/com/courtcanva/ccfranchise/jwts/JwtUsernameAndPasswordAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    }
+
+    @Override
+    @SneakyThrows
+    public Authentication attemptAuthentication(HttpServletRequest request,
+                                                HttpServletResponse response) throws AuthenticationException {
+
+        UsernameAndPasswordAuthenticationRequest authenticationRequest = new ObjectMapper()
+                .readValue(request.getInputStream(), UsernameAndPasswordAuthenticationRequest.class);
```
Comment (yuanningd) [inline src/main/java/com/courtcanva/ccfranchise/jwts/JwtUsernameAndPasswordAuthenticationFilter.java:43]: I think we have an easier way to get the user's name and password. you can check the `attemptAuthentication` implementation of `UsernamePasswordAuthenticationFilter` Class

File: src/main/java/com/courtcanva/ccfranchise/jwts/JwtUsernameAndPasswordAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        return authenticationManager.authenticate(authentication);
+    }
+
+    @Override
+    protected void successfulAuthentication(HttpServletRequest request,
+                                            HttpServletResponse response,
+                                            FilterChain chain, Authentication authResult) {
+
+        String token = Jwts.builder()
+                .setSubject(authResult.getName())
```
Comment (yuanningd) [inline src/main/java/com/courtcanva/ccfranchise/jwts/JwtUsernameAndPasswordAuthenticationFilter.java:59]: We better put the user id in the Jwt for future convenience

Comment (leonandhard) [reply src/main/java/com/courtcanva/ccfranchise/jwts/JwtUsernameAndPasswordAuthenticationFilter.java:62]: thx， add it

**Review by yuanningd — APPROVED** [review]

**Pull request merged**